code/launcher/xmake.lua

```diff
@@ -1,34 +1,28 @@
 target("Launcher")
     set_kind("phony")
-    
+
     on_install(function (target)
         local function run_command(cmd, args)
             print("Running command: " .. cmd .. " " .. table.concat(args or {}, " "))
             local stdout, stderr = os.iorunv(cmd, args)
         end
-    
+
         os.cd("code/launcher")
-    
+
         local npm = is_host("windows") and "npm.cmd" or "npm"
-    
+
         -- Run npm commands in the root directory
         run_command(npm, {"install"})
-        run_command(npm, {"run", "build"})
-    
-        -- Change to the electron directory
+        run_command(npm, {"run", "build:all"})
+
         local electron_dir = path.join(os.curdir(), "electron")
         if not os.isdir(electron_dir) then
             raise("Electron directory not found: " .. electron_dir)
         end
-        os.cd(electron_dir)
-    
-        -- Run npm commands in the electron directory
-        run_command(npm, {"install"})
-        run_command(npm, {"run", "build:electron"})
-    
+
         -- Copy produced files
         local src = path.join(electron_dir, "out", "win-unpacked")
-        local dst = path.join("..", "..", "..", target:installdir("launcher"))
```

Comment:
why go one less?

Reply:
This is because `npm run build:all` doesn't need to `cd` in electron directory. But this will be restored in the next PR to use `npm ci` per package instead.